common/app/navigation/Navigation.scala

```diff
@@ -0,0 +1,163 @@
+    // Football is currently in the News Pillar and the Sport pillar, however we don't want the parent to be News.
+    def isFootballInNews(parentTitle: String): Boolean = {
+      currentNavLink.title == "football" && parentTitle == "news"
+    }
+
+    @tailrec
+    def find(children: Seq[NavLink]): Option[NavLink] = {
+      children match {
+        case Nil => None
+        case head :: tail if (head == currentNavLink || head.children.contains(currentNavLink) && !isFootballInNews(head.title)) => Some(head)
+        case head :: tail => find(tail ++ head.children)
+      }
+    }
+    // If the parent isn't found within the current edition, check other editions
+    find(children ++ otherLinks).orElse(find(getChildrenFromOtherEditions(edition)))
+  }
+
+  def getPillar(currentParent: Option[NavLink], edition: Edition): Option[NavLink] = {
+    currentParent.flatMap( parent => {
+      if(otherLinks.contains(parent)) {
+        None
+      } else if (children.contains(parent)) {
+        currentParent
+      } else findParent(parent, edition).orElse(Some(ukNewsPillar))
+    })
+  }
+}
+
+object NavRoot {
+  def apply(edition: Edition): NavRoot = {
+    edition match {
+      case editions.Uk => NavRoot(Seq(ukNewsPillar, ukSportPillar, ukOpinionPillar, ukArtsPillar, ukLifestylePillar), ukOtherLinks, ukBrandExtensions)
+      case editions.Us => NavRoot(Seq(usNewsPillar, usSportPillar, usOpinionPillar, usArtsPillar, usLifestylePillar), usOtherLinks, usBrandExtensions)
+      case editions.Au => NavRoot(Seq(auNewsPillar, auSportPillar, auOpinionPillar, auArtsPillar, auLifestylePillar), auOtherLinks, auBrandExtensions)
+      case editions.International => NavRoot(Seq(auNewsPillar, auSportPillar, auOpinionPillar, auArtsPillar, auLifestylePillar), intOtherLinks, intBrandExtensions)
+    }
+  }
+}
+
+case class SimpleMenu private (root: NavRoot) {
+  def pillars: Seq[NavLink] = root.children
+  def otherLinks: Seq[NavLink] = root.otherLinks
+  def brandExtensions: Seq[NavLink] = root.brandExtensions
+}
+
+case class NavMenu private (page: Page, root: NavRoot, edition: Edition) {
+
+  def currentUrl: String = NavMenu.getSectionOrPageUrl(page, edition)
+  def pillars: Seq[NavLink] = root.children
+  def otherLinks: Seq[NavLink] = root.otherLinks
+  def brandExtensions: Seq[NavLink] = root.brandExtensions
+  def currentNavLink: Option[NavLink] = root.findDescendantByUrl(currentUrl, edition)
+  def currentParent: Option[NavLink] = currentNavLink.flatMap( link => root.findParent(link, edition) )
+  def currentPillar: Option[NavLink] = root.getPillar(currentParent, edition)
+  def subNavSections: Subnav = NavMenu.getSubnav(currentNavLink, currentParent, currentPillar)
+}
+
+object NavMenu {
+
+  def apply(page: Page, edition: Edition): NavMenu = NavMenu(page, NavRoot(edition), edition)
+  def apply(edition: Edition): SimpleMenu = SimpleMenu(NavRoot(edition))
+
+  def getSectionOrPageUrl(page: Page, edition: Edition): String = {
```

Comment:
you can make this function as well as `getSubnav` below package private (`private[navigation] def ...`)